### PR TITLE
fix(testing): align cfg_eval.model/callbacks with cfg_train in test_train_eval

### LINKED ISSUE #624: test(testing): test_train_eval needs cfg_eval.model + test/acc assertion aligned with ksin/ffn
## Context

Follow-up from [#623](https://github.com/tinaudio/synth-setter/pull/623) (which fixes #617 for the `FileNotFoundError` on `surge_mini`). While reviewing that PR, Copilot pointed out that the test's train->eval roundtrip is still incomplete even after aligning `cfg_eval.data` with `cfg_train.data`.

## Problems

After `cfg_eval.data = cfg_train.data` (ksin), `cfg_eval` still composes from `configs/eval.yaml` defaults:

- `model: surge_flow` -> `src.models.surge_flow_matching_module.SurgeFlowMatchingModule`
- `callbacks: eval_surge`

But `train(cfg_train)` produces an `ffn`/`KSinFeedForwardModule` checkpoint. In `src/eval.py::evaluate`, Lightning calls `trainer.test(model=model, ..., ckpt_path=cfg.ckpt_path)` which will attempt to load the ffn state_dict into the instantiated `surge_flow` model — this will fail on mismatched state_dict keys.

Additionally, `tests/test_eval.py:48` asserts:

```python
assert test_metric_dict["test/acc"] > 0.0
```

But neither `ksin_ff_module.test_step` nor `surge_flow_matching_module.test_step` logs `test/acc`. Only `mnist_module` does. The assertion is a stale artifact from when the default model was MNIST.

## Proposed fix

In `tests/test_eval.py::test_train_eval`:

1. Override `cfg_eval.model = cfg_train.model` and `cfg_eval.callbacks = cfg_train.callbacks` (or pick a matching eval-specific callbacks config) so the instantiated eval model matches the trained checkpoint.
2. Replace `test/acc` assertion with a metric both modules produce, e.g. `test/param_mse` or `test/loss` (assert it's a finite, non-negative float).

## Scope

- Scoped strictly to `tests/test_eval.py::test_train_eval`.
- Do not modify production configs (`configs/eval.yaml`, `configs/data/surge_mini.yaml`, model/callback configs).

## Parent

Part of #155 (Phase 2: Coverage & stability) -> #149 (Epic: test infrastructure & coverage).

Refs #623
Refs #617

### PR DESCRIPTION
**Pull request by ktinubu**
Fixes #624

Stacks on #623. That PR fixed only the `data` mismatch in `test_train_eval`; this PR completes the train<->eval alignment by also overriding `cfg_eval.model` and `cfg_eval.callbacks` to match `cfg_train`, so the evaluator loads the same ksin feedforward LightningModule and callback stack that training just checkpointed (instead of the default `surge_flow` / `eval_surge`).

Also replaces the stale `test/acc` assertions: `src/models/ksin_ff_module.py::test_step` logs `test/loss` (MSE), not `test/acc`. The new assertions use `test/loss < inf` as a sanity bound and keep a tight parity check between train-time test and standalone eval on the same checkpoint.

> **Edit:** the `test/loss < float("inf")` bound added in this PR is being tightened to `math.isfinite(test/loss)` in [#655](https://github.com/tinaudio/synth-setter/pull/655) — `<inf` silently accepts `-inf` and the stricter check rejects `+inf`, `-inf`, and `NaN`. Copilot flagged the gap while reviewing the sibling `test_train_validate` assertion in [#636](https://github.com/tinaudio/synth-setter/pull/636); both lines are swapped together in #655 to keep the canonical E2E template consistent. (The fix was initially committed on the #636 branch post-merge and had to be cherry-picked to a fresh branch in #655.)

When #623 lands in main, this PR should be rebased onto main; the diff will then show only the #624-specific delta.

## Test plan

- [x] `make format` clean
- [x] `pytest --collect-only tests/test_eval.py` still collects `test_train_eval`
- [x] GPU end-to-end: `pytest tests/test_eval.py::test_train_eval -v` PASSED in 489s on RTX 5060 Ti (CUDA 13.0)

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

All checks run against commit `307e495` on branch `fix/test-eval-model-callbacks-aligned` (worktree at `/home/build/synth-setter/.worktrees/fix-624-eval-alignment`, clean working tree).

- [x] **`test_train_eval` passes end-to-end on GPU with the new overrides and `test/loss` parity assertion** *(Level 1 — promise is end-to-end: train a ksin checkpoint, then `evaluate()` it with cfg_eval whose `data`/`model`/`callbacks` are overridden to match `cfg_train`, and confirm `test/loss` parity holds between train-time test-phase and standalone eval)*
  ```bash
  $ nvidia-smi --query-gpu=name,driver_version --format=csv,noheader
  NVIDIA GeForce RTX 5060 Ti, 580.126.09

  $ python3 -c "import torch; print('cuda:', torch.cuda.is_available(), 'devices:', torch.cuda.device_count())"
  cuda: True devices: 1

  $ time python3 -m pytest tests/test_eval.py::test_train_eval -v --no-header -o addopts=""
  collecting ... collected 1 item

  tests/test_eval.py::test_train_eval
  -------------------------------- live log call ---------------------------------
  WARNING  src.utils.instantiators:pylogger.py:46 [rank: 0] No logger configs found! Skipping...
  WARNING  src.utils.instantiators:pylogger.py:46 [rank: 0] No logger configs found! Skipping...
  PASSED                                                                   [100%]

  =================== 1 passed, 1 warning in 489.38s (0:08:09) ===================

  real    8m16.528s
  ```
  PASS in 489.38s (8m9s) on RTX 5060 Ti. A failure would mean either (a) Hydra resolution of the overrides broke, (b) the checkpoint's LightningModule mismatches `cfg_eval.model` and `load_from_checkpoint` rejects it, (c) `test/loss` isn't logged or parity is loose. None occurred.

- [x] **Test file contains exactly the four required edits (two config overrides + loss-based assertions, `test/acc` gone)** *(Level 5 — AST parse of `tests/test_eval.py` on the tested commit; confirms the Level 1 run exercised the NEW code path, not a stale one)*
  ```bash
  $ git rev-parse HEAD && git status --short
  307e495135f7c1aa6e1a90d74c10ba7ebadeb2c8

  $ python3 -c "
  import ast, pathlib
  src = pathlib.Path('tests/test_eval.py').read_text()
  tree = ast.parse(src)
  func = next(n for n in tree.body if isinstance(n, ast.FunctionDef) and n.name == 'test_train_eval')
  body_src = ast.unparse(func)
  checks = {
      'cfg_eval.data = cfg_train.data': 'cfg_eval.data = cfg_train.data' in body_src,
      'cfg_eval.model = cfg_train.model': 'cfg_eval.model = cfg_train.model' in body_src,
      'cfg_eval.callbacks = cfg_train.callbacks': 'cfg_eval.callbacks = cfg_train.callbacks' in body_src,
      'test/loss assertion present': 'test/loss' in body_src,
      'test/acc assertion absent': 'test/acc' not in body_src,
  }
  for k, v in checks.items():
      print(f'{\"OK\" if v else \"FAIL\"}: {k}')
  "
  OK: cfg_eval.data = cfg_train.data
  OK: cfg_eval.model = cfg_train.model
  OK: cfg_eval.callbacks = cfg_train.callbacks
  OK: test/loss assertion present
  OK: test/acc assertion absent
  ```

- [x] **Pytest still collects `test_train_eval` (RunIf + gpu marker do not gate it out on this host)** *(Level 2 — invokes pytest's collector, which is the tool that decides whether the test runs in CI; promise is narrower than a full run)*
  ```bash
  $ python3 -m pytest --collect-only tests/test_eval.py
  collected 1 item

  <Dir fix-624-eval-alignment>
    <Package tests>
      <Module test_eval.py>
        <Function test_train_eval>

  ========================== 1 test collected in 0.38s ===========================
  ```

- [ ] **Behavior on a GPU-less runner (e.g. standard GitHub Actions)** *(Level 1 — SKIP: this environment has a GPU so the `@RunIf(min_gpus=1)` gate cannot be exercised in the negative. The test is decorated with `@pytest.mark.gpu` and `@RunIf(min_gpus=1)`, so on a CPU-only runner pytest should skip it. To verify manually:)*
  ```bash
  # On a CPU-only machine (or CUDA_VISIBLE_DEVICES=""):
  CUDA_VISIBLE_DEVICES="" python3 -m pytest tests/test_eval.py::test_train_eval -v
  # Expect: SKIPPED (Requires: GPUs>=1)
  ```

### Summary

The core promise of this PR (`test_train_eval` passes with ksin-aligned `cfg_eval` + `test/loss` parity) is verified end-to-end on real hardware. The test is also confirmed to actually execute the new code path (edits present on the tested commit, clean working tree, pytest collects it).
